modules/gapi/test/gapi_gpu_test_kernels.cpp

```diff
@@ -0,0 +1,141 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+#include "test_precomp.hpp"
+
+#include <iomanip>
+#include "gapi_gpu_test_kernels.hpp"
+#include <opencv2/gapi/core.hpp>
+#include "opencl_kernels_test_gapi.hpp"
+
+
+namespace cv
+{
+#ifdef HAVE_OPENCL
+namespace gapi_test_kernels
+{
+
+GAPI_GPU_KERNEL(GGPUSymm7x7_test, TSymm7x7_test)
+{
+    static void run(const cv::UMat& in, cv::UMat &out)
+    {
+        if (cv::ocl::isOpenCLActivated())
+        {
+            cv::Size size = in.size();
+            size_t globalsize[2] = { (size_t)size.width, (size_t)size.height };
+
+            //size_t maxWorkItemSizes[32];
+            //cv::ocl::Device::getDefault().maxWorkItemSizes(maxWorkItemSizes);
+
+            cv::ocl::Kernel kernel;
+
+            int coefficients[10] = { 1140, -118, 526, 290, -236, 64, -128, -5, -87, -7 };
+            int shift = 10;
+
+
+            static const char * const borderMap[] = { "BORDER_CONSTANT", "BORDER_REPLICATE", "BORDER_UNDEFINED" };
+            std::string build_options = " -D BORDER_CONSTANT_VALUE=" + std::to_string(0) +
+                " -D " + borderMap[1] +
+                " -D SCALE=1.f/" + std::to_string(1 << shift) + ".f";
+
+
+            if (!kernel.create("symm_7x7_test", cv::ocl::gapi::symm7x7_test_oclsrc, build_options))
+            {
+                printf("symm_7x7_test OpenCL kernel creation failed with build_options = %s\n", build_options.c_str());
+            }
```

Comment:
So would kernel here be compiled only once or on every execution (`run()`)? cc: @alalek

Reply:
It will be compiled only once, I believe. Next time it will be retrieved from the cache.